### PR TITLE
refact: adopt personal pagerank & update ubuntu 20.04 & common 2.0.1

### LINKED ISSUE #136: 依赖的三方件log4j2版本2.13.3存在漏洞
### Bug Type (问题类型)

others (please edit later)

### Before submit

- [X] 我已经确认现有的 [Server-Issues](https://github.com/hugegraph/hugegraph/issues) 与 [Client-Issues](https://github.com/hugegraph/hugegraph-client/issues) 中没有相同 / 重复问题

### Environment (环境信息)

- Client Version: v1.9.6

### Expected & Actual behavior (期望与实际表现)

软件依赖的Apache log4j2版本2.13.3存在远程代码执行漏洞(CVE-2021-44228)。需要升级到2.15.0版本以上修复。

### Vertex/Edge example (问题点 / 边数据举例)

_No response_

### Schema [VertexLabel, EdgeLabel, IndexLabel] (元数据结构)

_No response_

### PR DESCRIPTION
**Pull request by imbajin**
maybe need also change commit id in shell

also fix #136

### CODE REVIEW

**Comment by javeme** [comment]
ci error:
```java
Running com.baidu.hugegraph.api.ApiTestSuite
Tests run: 148, Failures: 32, Errors: 116, Skipped: 0, Time elapsed: 3.085 sec <<< FAILURE!
com.baidu.hugegraph.api.PropertyKeyApiTest  Time elapsed: 1.864 sec  <<< ERROR!
com.baidu.hugegraph.rest.ClientException: Failed to do request
	at com.baidu.hugegraph.rest.AbstractRestClient.request(AbstractRestClient.java:168)
	at com.baidu.hugegraph.rest.AbstractRestClient.get(AbstractRestClient.java:265)
	at com.baidu.hugegraph.rest.AbstractRestClient.get(AbstractRestClient.java:239)
	at com.baidu.hugegraph.api.version.VersionAPI.get(VersionAPI.java:41)
	at com.baidu.hugegraph.driver.VersionManager.getApiVersion(VersionManager.java:45)
	at com.baidu.hugegraph.driver.HugeClient.checkServerApiVersion(HugeClient.java:99)
	at com.baidu.hugegraph.driver.HugeClient.initManagers(HugeClient.java:83)
	at com.baidu.hugegraph.driver.HugeClient.<init>(HugeClient.java:63)
	at com.baidu.hugegraph.driver.HugeClientBuilder.build(HugeClientBuilder.java:67)
	at com.baidu.hugegraph.BaseClientTest.open(BaseClientTest.java:45)
	at com.baidu.hugegraph.BaseClientTest.init(BaseClientTest.java:52)
	at com.baidu.hugegraph.api.BaseApiTest.init(BaseApiTest.java:72)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

com.baidu.hugegraph.api.PropertyKeyApiTest  Time elapsed: 1.868 sec  <<< FAILURE!
java.lang.AssertionError: Not opened client
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertNotNull(Assert.java:712)
	at com.baidu.hugegraph.api.BaseApiTest.clear(BaseApiTest.java:98)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,8 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-16.04
+    # which version is stable?
+    runs-on: ubuntu-latest
```
Comment (javeme) [inline .github/workflows/ci.yml]: keep the same with server version ubuntu-20.04 ?

**Comment by javeme** [comment]
ci error:
```java
Running com.baidu.hugegraph.functional.FuncTestSuite
com.baidu.hugegraph.exception.InvalidOperationException: The vertex '196:vadas' doesn't have the property 'not-exist'
com.baidu.hugegraph.exception.InvalidOperationException: The edge 'S242:peter>168>>S243:lop' doesn't have the property 'not-exist'
com.baidu.hugegraph.exception.InvalidOperationException: Must set source vertex id or add vertices before add edges
com.baidu.hugegraph.exception.InvalidOperationException: Must set source vertex id or add vertices before add edges
Tests run: 100, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 244.509 sec <<< FAILURE!
testHttpsClientNewBuilderZeroPoolParam(com.baidu.hugegraph.functional.HugeClientHttpsTest)  Time elapsed: 10.064 sec  <<< ERROR!
com.baidu.hugegraph.rest.ClientException: Failed to do request
	at com.baidu.hugegraph.rest.AbstractRestClient.request(AbstractRestClient.java:168)
	at com.baidu.hugegraph.rest.AbstractRestClient.get(AbstractRestClient.java:265)
	at com.baidu.hugegraph.rest.AbstractRestClient.get(AbstractRestClient.java:239)
	at com.baidu.hugegraph.api.version.VersionAPI.get(VersionAPI.java:41)
	at com.baidu.hugegraph.driver.VersionManager.getApiVersion(VersionManager.java:45)
	at com.baidu.hugegraph.driver.HugeClient.checkServerApiVersion(HugeClient.java:99)
	at com.baidu.hugegraph.driver.HugeClient.initManagers(HugeClient.java:83)
	at com.baidu.hugegraph.driver.HugeClient.<init>(HugeClient.java:63)
	at com.baidu.hugegraph.driver.HugeClientBuilder.build(HugeClientBuilder.java:67)
	at com.baidu.hugegraph.functional.HugeClientHttpsTest.testHttpsClientNewBuilderZeroPoolParam(HugeClientHttpsTest.java:118)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.glassfish.jersey.apache.connector.ApacheConnector.apply(ApacheConnector.java:435)
	... 62 more
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:457)
	at sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:165)
	at sun.security.ssl.SSLTransport.decode(SSLTransport.java:109)
	... 77 more


Results :

Tests in error: 
  testHttpsClientNewBuilderZeroPoolParam(com.baidu.hugegraph.functional.HugeClientHttpsTest): Failed to do request
  testHttpsClientWithConnectionPoolNoTimeOutParam(com.baidu.hugegraph.functional.HugeClientHttpsTest): Failed to do request

Tests run: 100, Failures: 0, Errors: 2, Skipped: 0
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (imbajin) [reply .github/workflows/ci.yml]: 18, 20, latest can't work

**Comment by javeme** [comment]
```java
Tests run: 371, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 197.989 sec <<< FAILURE!
testPersonalRankWithDegree(com.baidu.hugegraph.api.traverser.PersonalRankApiTest)  Time elapsed: 0.02 sec  <<< ERROR!
class java.lang.IllegalArgumentException: The max depth of rank request must be in range (1, 50], but got '1'
	at com.baidu.hugegraph.exception.ServerException.fromResponse(ServerException.java:47)
	at com.baidu.hugegraph.client.RestClient.checkStatus(RestClient.java:93)
	at com.baidu.hugegraph.rest.AbstractRestClient.post(AbstractRestClient.java:200)
	at com.baidu.hugegraph.rest.AbstractRestClient.post(AbstractRestClient.java:174)
	at com.baidu.hugegraph.api.traverser.PersonalRankAPI.post(PersonalRankAPI.java:41)
	at com.baidu.hugegraph.api.traverser.PersonalRankApiTest.testPersonalRankWithDegree(PersonalRankApiTest.java:168)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)


Results :

Tests in error: 
  testPersonalRankWithDegree(com.baidu.hugegraph.api.traverser.PersonalRankApiTest): The max depth of rank request must be in range (1, 50], but got '1'

Tests run: 371, Failures: 0, Errors: 1, Skipped: 0
```

**Comment by javeme** [comment]
error when running on local mac:
```java
GRIZZLY0013: Exception during FilterChain execution
org.glassfish.grizzly.ssl.SSLContextConfigurator$GenericStoreException: java.io.FileNotFoundException: conf/hugegraph-server.keystore (No such file or directory)
	at org.glassfish.grizzly.ssl.SSLContextConfigurator.createSSLContext(SSLContextConfigurator.java:506)
	at org.glassfish.grizzly.ssl.SSLEngineConfigurator.createSSLEngine(SSLEngineConfigurator.java:185)
	at org.glassfish.grizzly.ssl.SSLEngineConfigurator.createSSLEngine(SSLEngineConfigurator.java:162)
	at org.glassfish.grizzly.ssl.SSLBaseFilter$SSLTransportFilterWrapper.handleRead(SSLBaseFilter.java:1112)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:515)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
	at org.glassfish.grizzly.strategies.SameThreadIOStrategy.executeIoEvent(SameThreadIOStrategy.java:79)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.executeIoEvent(AbstractIOStrategy.java:66)
	at org.glassfish.grizzly.nio.SelectorRunner.iterateKeyEvents(SelectorRunner.java:391)
	at org.glassfish.grizzly.nio.SelectorRunner.iterateKeys(SelectorRunner.java:360)
	at org.glassfish.grizzly.nio.SelectorRunner.doSelect(SelectorRunner.java:324)
	at org.glassfish.grizzly.nio.SelectorRunner.run(SelectorRunner.java:255)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.FileNotFoundException: conf/hugegraph-server.keystore (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at org.glassfish.grizzly.ssl.SSLContextConfigurator.loadBytes(SSLContextConfigurator.java:641)
	at org.glassfish.grizzly.ssl.SSLContextConfigurator.createSSLContext(SSLContextConfigurator.java:476)
	... 20 more
```

**Comment by javeme** [comment]
```java
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.baidu.hugegraph.api.ApiTestSuite
Tests run: 148, Failures: 32, Errors: 116, Skipped: 0, Time elapsed: 4.177 sec <<< FAILURE!
com.baidu.hugegraph.api.PropertyKeyApiTest  Time elapsed: 2.369 sec  <<< ERROR!
com.baidu.hugegraph.rest.ClientException: Failed to do request
	at com.baidu.hugegraph.rest.AbstractRestClient.request(AbstractRestClient.java:199)
	at com.baidu.hugegraph.rest.AbstractRestClient.get(AbstractRestClient.java:296)
	at com.baidu.hugegraph.rest.AbstractRestClient.get(AbstractRestClient.java:270)
	at com.baidu.hugegraph.api.version.VersionAPI.get(VersionAPI.java:41)
	at com.baidu.hugegraph.driver.VersionManager.getApiVersion(VersionManager.java:45)
	at com.baidu.hugegraph.driver.HugeClient.checkServerApiVersion(HugeClient.java:99)
	at com.baidu.hugegraph.driver.HugeClient.initManagers(HugeClient.java:83)
	at com.baidu.hugegraph.driver.HugeClient.<init>(HugeClient.java:63)
	at com.baidu.hugegraph.driver.HugeClientBuilder.build(HugeClientBuilder.java:67)
	at com.baidu.hugegraph.BaseClientTest.open(BaseClientTest.java:45)
	at com.baidu.hugegraph.BaseClientTest.init(BaseClientTest.java:52)
	at com.baidu.hugegraph.api.BaseApiTest.init(BaseApiTest.java:72)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: javax.ws.rs.ProcessingException: org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:8080 [/127.0.0.1] failed: Connection refused (Connection refused)
	at org.glassfish.jersey.apache.connector.ApacheConnector.apply(ApacheConnector.java:481)
	at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:255)
	at org.glassfish.jersey.client.JerseyInvocation$1.call(JerseyInvocation.java:684)
	at org.glassfish.jersey.client.JerseyInvocation$1.call(JerseyInvocation.java:681)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:228)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:444)
	at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:681)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:411)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.get(JerseyInvocation.java:311)
	at com.baidu.hugegraph.rest.AbstractRestClient.lambda$get$3(AbstractRestClient.java:301)
	at com.baidu.hugegraph.rest.AbstractRestClient.request(AbstractRestClient.java:197)
	... 41 more
Caused by: org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:8080 [/127.0.0.1] failed: Connection refused (Connection refused)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:151)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:380)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
	at org.glassfish.jersey.apache.connector.ApacheConnector.apply(ApacheConnector.java:435)
	... 53 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:607)
	at org.apache.http.conn.socket.PlainConnectionSocketFactory.connectSocket(PlainConnectionSocketFactory.java:74)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:134)
	... 62 more
```

File: src/main/java/com/baidu/hugegraph/version/ClientVersion.java [hunk 2]
```diff
@@ -33,9 +33,9 @@
 
     public static final Version VERSION = Version.of(ClientVersion.class);
 
-    public static final void check() {
+    public static void check() {
         // Check version of hugegraph-common
-        VersionUtil.check(CommonVersion.VERSION, "1.8", "1.9",
+        VersionUtil.check(CommonVersion.VERSION, "1.8", "2.1",
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/version/ClientVersion.java:38]: "2.0", "2.1"

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -14,10 +14,10 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-16.04
+    runs-on: ubuntu-20.04
     env:
       TRAVIS_DIR: assembly/travis
-      COMMIT_ID: 0dc7b271835e53136ee92691b07fa4abe925e16e
+      COMMIT_ID: d213166eb2185cbb6bd8bd5d201b7167e7b98993
```
Comment (javeme) [inline .github/workflows/ci.yml:20]: revert to the previous commit,
and adopt can be update to zulu

**Review by javeme — APPROVED** [review]

File: src/test/java/com/baidu/hugegraph/functional/HugeClientHttpsTest.java [hunk 4]
```diff
@@ -65,6 +66,7 @@ public void testHttpsClientBuilderWithConnection() {
     @Test
     public void testHttpsClientWithConnectionPoolNoUserParam() {
         client = HugeClient.builder(BASE_URL, GRAPH)
+                           .configUser(USERNAME, PASSWORD)
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/functional/HugeClientHttpsTest.java:69]: maybe should not add username for test case testHttpsClientWithConnectionPoolNoUserParam

**Pull request merged**